src/pcre2_compile.c

```diff
@@ -10813,7 +10856,7 @@ re->blocksize = re_blocksize;
 re->magic_number = MAGIC_NUMBER;
 re->compile_options = options;
 re->overall_options = cb.external_options;
-re->extra_options = ccontext->extra_options;
+re->extra_options = xoptions;
```

Comment:
Cannot we simply modify this flag directly?

Reply:
No, the pattern can't modify the compile-context, which could be reused with another pattern.